actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -691,7 +691,7 @@ def optimize_routes_generation?
       end
 
       def find_script_name(options)
-        options.delete(:script_name) || ''
+        options.delete(:script_name) || ActionController::Base.config.relative_url_root || ''
```

Comment:
I'm not sure if we should be coupling Action Dispatch with Action Controller here. Do we have any precedent? cc @pixeltrix 

Can we inject this config at load time using the railtie?